agent/connect/uri_service.go

```diff
@@ -35,3 +32,20 @@ func (id SpiffeIDService) URI() *url.URL {
 	result.Path = id.uriPath()
 	return &result
 }
+
+func (id SpiffeIDService) uriPath() string {
+	path := fmt.Sprintf("/ns/%s/dc/%s/svc/%s",
+		id.NamespaceOrDefault(),
+		id.Datacenter,
+		id.Service,
+	)
+
+	// Although OSS has no support for partitions, it still needs to be able to
+	// handle exportedPartition from peered Consul Enterprise clusters in order
+	// to generate the correct SpiffeID.
+	// We intentionally avoid using pbpartition.DefaultName here to be OSS friendly.
+	if ap := id.PartitionOrDefault(); ap != "" && ap != "default" {
+		return "/ap/" + ap + path
+	}
+	return path
+}
```

Comment:
Sanity check: is there a reason we don't want /ap/ in OSS, side from "partitions are enterprise-only"? Wondering if there are side effects

Reply:
Backwards compatibility. We defined the `/ap` prefix as "only present if not the default"